src/main/java/net/dv8tion/jda/core/requests/restaction/RoleAction.java

```diff
@@ -85,15 +82,20 @@ public RoleAction setColor(Integer rgb)
 
     public RoleAction setPermissions(Permission... permissions)
     {
-        Args.notNull(permissions, "Permissions");
-        this.permissions = new HashSet<>();
-        Collections.addAll(this.permissions, permissions);
+        this.permissions = permissions == null ? 0 : Permission.getRaw(permissions);
         return this;
     }
 
     public RoleAction setPermissions(Collection<Permission> permissions)
     {
-        this.permissions = permissions == null ? null : new HashSet<>(permissions);
+        this.permissions = permissions == null ? 0 : Permission.getRaw(permissions);
+        return this;
+    }
+
+    public RoleAction setPermissions(long permissions)
+    {
+        Args.notNegative(permissions, "Raw Permissions");
+        this.permissions = permissions;
```

Comment:
Should probably also make sure that the provided long isn't larger than `Permission.getRaw(Permission.values())`